backend/src/routes/events.ts

```diff
@@ -39,7 +39,24 @@ type ServerSideEventData = WatchEvent | SettingsEvent | { type: 'START' | 'LOADE
 
 let requests: { cancel: () => void }[] = []
 
-const resourceCache: {
+export function getKubeResources(kind: string, apiVersion: string) {
+  const option = { apiVersion, kind }
+  const apiVersionPlural = apiVersionPluralFn(option)
+  return Object.values(resourceCache[apiVersionPlural] || {}).map((event) => {
+    return event.resource
+  })
+}
+
+export async function isAuth(resource: IResource, token: string): Promise<boolean> {
+  let allowed = await canListClusterScopedKind(resource, token)
+  if (allowed) return true
+  allowed = await canListNamespacedScopedKind(resource, token)
+  if (allowed) return true
+  allowed = await canGetResource(resource, token)
+  return allowed
```

Comment:
This looks correct for local-cluster resources, but it is not correct for managed cluster resources. For managed clusters, we need to check if the user can create ManagedClusterView in the managed cluster namespace.

Compare what the `e2e-view-ns` user can see on the weekly cluster.

For OCP apps, I think at this point we have a faked resource, usually the Deployment. Does this vary? For local-cluster OCP apps, we need to think about what happens if the user can see some but not all of the resources.

Reply:
fixed

Reply:
This needs to check if the resource can be created in the namespace for a specific managed cluster. For example, I created a `Role` and `RoleBinding` to grant `e2e-admin-ns` access to create a `ManagedClusterView` in the `weekly-managed` namespace. This should allow the user to see apps from that cluster, but your code is only checking if the user can create this resource in _any_ namespace.

Reply:
done